packages/cli/cloud/src/create-project/utils/project-questions.ts

```diff
@@ -0,0 +1,73 @@
+import { DistinctQuestion } from 'inquirer';
+import type { ProjectAnswers } from '../../types';
+
+/**
+ * Apply default values to questions based on the provided mapper
+ * @param QuestionsMapper - A partial object with keys matching the ProjectAnswers keys and values being the default value or a function to get the default value
+ */
+export function questionDefaultValuesMapper(
+  QuestionsMapper: Partial<{
+    [K in keyof ProjectAnswers]:
+      | ((question: DistinctQuestion<ProjectAnswers>) => ProjectAnswers[K])
+      | ProjectAnswers[K];
+  }>
+) {
+  return (
+    questions: ReadonlyArray<DistinctQuestion<ProjectAnswers>>
```

Comment:
Are we really expecting to apply the default values to various sets of questions?
If not, it might be argued that a pure function would be more suitable (for simplicity & readability).

Reply:
> Are we really expecting to apply the default values to various sets of questions?
As I was there, I made a generic function that could handle all the questions.
It's the second time we have had this need (the first was the project's name), so we are not excluding having this need again in future.

I don't really understand why a "simple function" (I guess you mean only one level function that is getting everything in parameters) would enhance readability. Wouldn't it be the same if we curry the function you talk about?
For me, splitting into multiple functions (having a context for each, first for setup, second for mapping) is more readable (and simpler, with fewer parameters and a clear purpose) than having too many parameters for a single function.

Reply:
Hmm then what's the limit? If there are 2 parameter, should we go for higher order function systematically?
We could end up with a bit of overengineering I feel like ^^' 

But I understand your point, let's keep it as is 👍